libfreerdp/codec/dsp.c

```diff
@@ -768,9 +772,15 @@ static BOOL freerdp_dsp_encode_ima_adpcm(FREERDP_DSP_CONTEXT* context, const BYT
 			*dst++ = encoded;
 			size -= 4;
 		}
+
+		if (dst - start == context->adpcm.ima.packet_size)
+		{
+			Stream_Write(out, start, context->adpcm.ima.packet_size);
+			dst = Stream_Buffer(context->buffer);
+		}
 	}
 
-	Stream_SetPointer(out, dst);
+	Stream_SetPointer(context->buffer, dst);
```

Comment:
not really sure what you try to do with this line.

Reply:
Maybe I need to better explain what the issue was, and what I did to fix it.
Before, this encoding function was in JIT production without buffering. It was working before the deep audin refactoring (those who decouple backend and dsp) because the input of this function had already been properly sized.
But now, this function gets data directly from the backend, and the backend does not guarantee that the data has always the same and the good size.
Since the protocol needs an alignment and a minimal frame number per packet, I have added some buffering here to ensure we output properly sized packets.

So here at the end of the function, I do not modify the output buffer, but only the intermediate buffer (that contain well-encoded data, but does not contain enough data to be sent yet)

Reply:
See comment above